include/Oscillator.h

```diff
@@ -163,7 +163,7 @@ class LMMS_EXPORT Oscillator
 
 	static inline sample_t noiseSample( const float )
 	{
-		return 1.0f - rand() * 2.0f / RAND_MAX;
+		return 1.0f - rand() * 2.0f / static_cast<float>(RAND_MAX);
```

Comment:
I wonder if it would be worth adding a range version for `rand` like @rubiefawn did for `fastRand`:

https://github.com/LMMS/lmms/blob/3417dfe86da0f6e9a41b3e71e30485a84213ffbf/include/lmms_math.h#L88-L99

Reply:
That might make sense too, but I am focusing more on #7677 rn so this has to wait a bit, if i'm refactoring

Reply:
There's a lot of places in the code where `rand()`/`std::rand()` is used in this way, which we may want to change to just use `fastRand()`. I would argue that belongs in a separate PR since it is beyond the scope of "fix compiler warnings".